### PR TITLE
feat(module:image): add image component

### PR DESCRIPTION
**Pull request by stygian-desolator**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

add image component

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by stygian-desolator** [comment]
@hsuanxyz cc, thanks~

File: components/image/image-preview.component.ts [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+  isDragging = false;
+  visible = true;
+  animationState: 'void' | 'enter' | 'leave' = 'enter';
+  animationStateChanged = new EventEmitter<AnimationEvent>();
+
+  previewImageTransform = '';
+  previewImageWrapperTransform = '';
+  operations: NzImageContainerOperation[] = [
+    {
+      icon: 'close',
```
Comment (hsuanxyz) [inline components/image/image-preview.component.ts:117]: The icons here need to be imported statically in `components/icon/icons.ts`

File: components/image/doc/index.en-US.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+| nzZoom      | Zoom rate | `number` | 1 |
+| nzRotate      | Rotate rate | `number` | 0 |
+
+### NzImagePreviewRef
+| Property | Description | Type | Default |
+| --- | --- | --- | --- |
+| active | Set active index | `(index: number) => void` | - |
+| prev | Previous image | `() => void` | - |
+| next | Next image | `() => void` | - |
+| close | Close image preview | `() => void` | - |
```
Comment (hsuanxyz) [inline components/image/doc/index.en-US.md:60]:
```suggestion
| Name | Description |
| --- | --- |
| active(index: number): void | Set active index |
| prev(): void | Previous image |
| next(): void | Next image |
| close(): void | Close image preview |
```

File: components/image/doc/index.zh-CN.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+| nzZoom      | 缩放比例 | `number` | 1 |
+| nzRotate      | 旋转角度 | `number` | 0 |
+
+### NzImagePreviewRef
+| 参数 | 说明 | 类型 | 默认值 |
+| --- | --- | --- | --- |
+| active | 设置预览索引 | `(index: number) => void` | - |
+| prev | 上一张 | `() => void` | - |
+| next | 下一张 | `() => void` | - |
+| close | 关闭预览 | `() => void` | - |
```
Comment (hsuanxyz) [inline components/image/doc/index.zh-CN.md:60]:
```suggestion
| 名称 | 描述 |
| --- | --- |
| active(index: number): void | 设置预览索引 |
| prev(): void | 上一张 |
| next(): void | 下一张 |
| close(): void | 关闭预览 |
```

**Review by hsuanxyz — CHANGES_REQUESTED** [review]
Good work!

**Comment by stygian-desolator** [comment]
@hsuanxyz  Please review again, thanks~

**Comment by hsuanxyz** [comment]
@stygian-desolator The fit-content action is only applied when the visible area is not completely covered.

image

**Pull request merged**